960-api-services/src/main/java/software/wings/service/impl/AwsApiHelperService.java

```
@@ -751,4 +761,41 @@ private static void sortDescending(List<S3ObjectSummary> objectSummaryList) {
 
     objectSummaryList.sort((o1, o2) -> o2.getLastModified().compareTo(o1.getLastModified()));
   }
+  private void handleExceptionWhileFetchingRepositories(int code, String errormessage) {
+    switch (code) {
+      case 404:
+        throw NestedExceptionUtils.hintWithExplanationException(
```

If I understand correctly from your comments in Slack we only see the hint part. If that is the case, lets throw a regular exception instead of Hint/Explanation. The exception can be a single string that has both reason, explanation & hint as to what can be done to get rid of the message.